internal/util/identifiers.go

```diff
@@ -15,9 +15,13 @@ import (
 	"go.uber.org/zap"
 )
 
+// deterministicRand is used while running tests to ensure that the generated
+// identifiers are deterministic.
```

Comment:
Maybe add a note that this is reset each test scenario because we restart the backend for each.